builtin/plugins/dposv3/dpos.go

```diff
@@ -537,6 +536,11 @@ func (c *DPOS) CheckRewardsFromAllValidators(ctx contract.StaticContext, req *Ch
 	if req.Delegator == nil {
 		return nil, logStaticDposError(ctx, errors.New("CheckRewardsFromAllValidators called with req.Delegator == nil"), req.String())
 	}
+
+	if ctx.FeatureEnabled(features.DPOSVersion3_6, false) {
+		return c.checkRewardsFromAllValidators2(ctx, req)
```

Comment:
This is a static method that isn't called from any non-static methods, so the feature check is unnecessary, you can just modify the original method.